scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -300,6 +302,10 @@ trait TypedPipe[+T] extends Serializable {
   def mapValues[K, V, U](f: V => U)(implicit ev: T <:< (K, V)): TypedPipe[(K, U)] =
     raiseTo[(K, V)].map { case (k, v) => (k, f(v)) }
 
+  /** Similar to mapValues, but allows to return a collection of outputs for each input value */
+  def flatMapValues[K, V, U](f: V => TraversableOnce[U])(implicit ev: T <:< (K, V)): TypedPipe[(K, TraversableOnce[U])] =
```

Comment:
I think we want `TypedPipe[(K, U)]` to be the return type. In other words, you have another copy of mapValues here, which you can use if you don't want the flattening, but I think this one should behave exactly as it does on KeyedListLike.